src/domain/routes/index.types.d.ts

```diff
@@ -0,0 +1,27 @@
+type PayViaRoutesArgs = import("lightning").PayViaRoutesArgs
+type RawRoute = { safe_fee: number } & PayViaRoutesArgs["routes"][number]
+type Route = { roundedUpFee: Satoshis; roundedDownFee: Satoshis }
+
+type CachedRoute = { pubkey: Pubkey; route: RawRoute }
+
+declare const cacheKeySymbol: unique symbol
+type CacheKey = string & { [cacheKeySymbol]: never }
+
+type CachedRouteKeyGenerator = {
+  generate(milliSatsAmounts: MilliSatoshis): CacheKey
+}
```

Comment:
has this the same problem discussed for fee calculators?

Reply:
Ah do you mean we should also pass the `paymentHash` into the method instead of the constructor... I don't mind that change if yes

Reply:
Changed here: https://github.com/GaloyMoney/galoy/pull/594/commits/6efb0abebf31ef6c97c4c57affc220429ba266d2